### PR TITLE
Don't output the whole Rails::Railtie object when a NoMethodError is raised

### PR DESCRIPTION
**Pull request by Edouard-chin**
Don't output the whole Rails::Railtie object when a NoMethodError is raised

- ### Summary

  The terminal output on a Rails application running ruby 3 will be
  cluttered with thousands of lines if one inadventarly call a
  unexisting method.
  This happen in various places (IntegrationTest, when running a db
  migration ...).
  This is related to a change in ruby 3 when a NoMethodError is
  raised.

  ### Simple reproduction

  ```ruby
  class A
    def initialize(session)
      @A = session
    end
  end

  test = A.new("*" * 36)
  test.dsad # undefined method `dsad' for #<A:0x00007f847d8494b0 @A="************************************"> (NoMethodError)
  # Note that the "#<A:0x00007f847d8494b0 @A="************************************">" part is 65 chars long.

  test = test = A.new("*" * 37)
  test.dsad # undefined method `dsad' for #<A:0x00007fa8c38299c0> (NoMethodError)
  ```

  On Ruby < 3, the NoMethodError message (everything starting from the
  "#" char) could only be 65 characters long. If it was above that
  ruby would only output the name of the class and its address.

  On Ruby >= 3, that limitation has been removed and the message can
  be any length long.

  Change upstream was here https://github.com/ruby/ruby/commit/1258a0fb90ea63bf1ec108815cce0d552acfc726

  ### On Rails

  Anytime a method is called on a object that holds the entire
  Rails::Application, the terminal would output the entire application
  which is annoying and be can be dangerous because it will leak
  everything containing the credentials (stored inside the Application
  object).

### CODE REVIEW

**Comment by jonathanhefner** [comment]
I noticed this too when typing e.g. `Rails.application.config.does_not_exist` in the console.  What do you think about reducing the output of `Application#inspect` instead?

**Comment by Edouard-chin** [comment]
Make sense 👍 , just noticed this behaviour in many more places . Will change

**Comment by Edouard-chin** [comment]
Done, decided to override the inspect method on the Railtie object because any engine would have this problem as well. Also updated my explanation on why this is happening because it wasn't very accurate.

File: railties/test/engine_test.rb [hunk 1]
```diff
@@ -24,4 +24,14 @@ def initialize
     end
     assert_equal "world", klass.instance.hello
   end
+
+  test "application object isn't output when a NoMethorError is raised" do
```
Comment (gmcgibbon) [inline railties/test/engine_test.rb:28]: Shouldn't we be testing this at the railtie level?

Comment (rafaelfranca) [reply railties/test/engine_test.rb:28]: It is checking with an application though, I think we need to either use an engine or a railtie and move this file in this case.

**Comment by Edouard-chin** [comment]
Thanks for the reviews, I tested on a Railtie instead and moved the test in a more appropriate test file

**Pull request merged**